### PR TITLE
C#: Improve GetFiles in the Dependency Manager.

### PR DESCRIPTION
**Pull request by michaelnebel**
In this PR we restrict the files that is scanned for packages references and ASP.NET framework dependency to files smaller than 1 MB. Furthermore, we fetch the source dir file info fewer times.
On my machine this had a positive performance impact, especially when source directory contains compressed files and large log files.

### CODE REVIEW

File: csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs [hunk 1]
```diff
@@ -374,14 +388,19 @@ private void Restore(IEnumerable<string> targets, string? pathToNugetConfig = nu
             }
         }
 
-        private void DownloadMissingPackages()
+        private void DownloadMissingPackages(List<FileInfo> allFiles)
         {
-            var nugetConfigs = GetFiles("nuget.config", recurseSubdirectories: true).ToArray();
+            var nugetConfigs = allFiles
+                .Where(fi => fi.Name == "nuget.config")
```
Comment (tamasvajk) [inline csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs:394]: Shouldn't we check equality with case insensitivity?

File: csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs [hunk 2]
```diff
@@ -374,14 +388,19 @@ private void Restore(IEnumerable<string> targets, string? pathToNugetConfig = nu
...
+                .Select(fi => fi.FullName)
+                .ToArray();
             string? nugetConfig = null;
             if (nugetConfigs.Length > 1)
             {
                 progressMonitor.MultipleNugetConfig(nugetConfigs);
-                nugetConfig = GetFiles("nuget.config", recurseSubdirectories: false).FirstOrDefault();
+                nugetConfig = GetFiles("nuget.config", recurseSubdirectories: false)
```
Comment (tamasvajk) [inline csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs:401]: We could probably save this `GetFiles` call too by checking if the file name is `Path.Combine(sourceDir,"nuget.config")`.

File: csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs [hunk 3]
```diff
@@ -185,15 +184,30 @@ private void GenerateSourceFilesFromWebViews()
 
         public DependencyManager(string srcDir) : this(srcDir, DependencyOptions.Default, new ConsoleLogger(Verbosity.Info)) { }
 
-        private IEnumerable<string> GetFiles(string pattern, bool recurseSubdirectories = true) =>
+        private IEnumerable<FileInfo> GetFiles(string pattern, bool recurseSubdirectories = true) =>
```
Comment (tamasvajk) [inline csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs:187]: This method could probably be simplified to `GetAllFiles()` if `GetFiles("nuget.config", recurseSubdirectories: false)` is refactored.

**Review by tamasvajk — COMMENTED** [review]
Looks good, I've added some minor comments.

**Review by tamasvajk — APPROVED** [review]

**Comment by michaelnebel** [comment]
DCA looks good.

**Pull request merged**